File: Chess/v1/Chess.java
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

public class Chess extends JFrame {
    public Chess() {
        add(new ChessPanel());
        setTitle("Simple Java Chess");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setResizable(false);
        pack();
        setLocationRelativeTo(null);
        setVisible(true);
    }

    public static void main(String[] args) {
        new Chess();
    }
}

class ChessPanel extends JPanel implements MouseListener {
    private final int TILE = 80;
    private String[][] board = new String[8][8];
    private int selectedRow = -1;
    private int selectedCol = -1;
    private boolean whiteTurn = true;
    private String message = "White to move";

    public ChessPanel() {
        setPreferredSize(new Dimension(640, 700));
        addMouseListener(this);
        setupBoard();
    }

    private void setupBoard() {
        board[0] = new String[]{"bR","bN","bB","bQ","bK","bB","bN","bR"};
        board[1] = new String[]{"bP","bP","bP","bP","bP","bP","bP","bP"};
        for (int r = 2; r < 6; r++) {
            for (int c = 0; c < 8; c++) board[r][c] = "";
        }
        board[6] = new String[]{"wP","wP","wP","wP","wP","wP","wP","wP"};
        board[7] = new String[]{"wR","wN","wB","wQ","wK","wB","wN","wR"};
    }

    protected void paintComponent(Graphics g) {
        super.paintComponent(g);

        for (int r = 0; r < 8; r++) {
            for (int c = 0; c < 8; c++) {
                if ((r + c) % 2 == 0) g.setColor(new Color(240, 217, 181));
                else g.setColor(new Color(181, 136, 99));

                g.fillRect(c * TILE, r * TILE, TILE, TILE);

                if (r == selectedRow && c == selectedCol) {
                    g.setColor(Color.YELLOW);
                    g.drawRect(c * TILE + 3, r * TILE + 3, TILE - 6, TILE - 6);
                    g.drawRect(c * TILE + 4, r * TILE + 4, TILE - 8, TILE - 8);
                }

                drawPiece(g, board[r][c], c * TILE, r * TILE);
            }
        }

        g.setColor(Color.BLACK);
        g.fillRect(0, 640, 640, 60);
        g.setColor(Color.WHITE);
        g.setFont(new Font("Arial", Font.BOLD, 22));
        g.drawString(message, 20, 675);
    }

    private void drawPiece(Graphics g, String p, int x, int y) {
        if (p.equals("")) return;

        String symbol = p.substring(1);
        g.setFont(new Font("Arial", Font.BOLD, 42));
        g.setColor(p.charAt(0) == 'w' ? Color.WHITE : Color.BLACK);
        g.drawString(symbol, x + 25, y + 52);
    }

    public void mousePressed(MouseEvent e) {
        int col = e.getX() / TILE;
        int row = e.getY() / TILE;

        if (row < 0 || row >= 8 || col < 0 || col >= 8) return;

        if (selectedRow == -1) {
            if (!board[row][col].equals("") && isCorrectTurn(board[row][col])) {
                selectedRow = row;
                selectedCol = col;
            }
        } else {
            if (isValidMove(selectedRow, selectedCol, row, col)) {
                board[row][col] = board[selectedRow][selectedCol];
                board[selectedRow][selectedCol] = "";
                whiteTurn = !whiteTurn;
                message = whiteTurn ? "White to move" : "Black to move";
            } else {
                message = "Illegal move";
            }

            selectedRow = -1;
            selectedCol = -1;
        }

        repaint();
    }

    private boolean isCorrectTurn(String piece) {
        return whiteTurn && piece.charAt(0) == 'w' || !whiteTurn && piece.charAt(0) == 'b';
    }

    private boolean isValidMove(int sr, int sc, int er, int ec) {
        String piece = board[sr][sc];
        if (piece.equals("")) return false;

        String target = board[er][ec];
        if (!target.equals("") && target.charAt(0) == piece.charAt(0)) return false;

        int dr = er - sr;
        int dc = ec - sc;
        char color = piece.charAt(0);
        char type = piece.charAt(1);

        if (type == 'P') {
            int dir = color == 'w' ? -1 : 1;
            int start = color == 'w' ? 6 : 1;

            if (dc == 0 && dr == dir && target.equals("")) return true;
            if (dc == 0 && sr == start && dr == 2 * dir && target.equals("") && board[sr + dir][sc].equals("")) return true;
            if (Math.abs(dc) == 1 && dr == dir && !target.equals("")) return true;
        }

        if (type == 'R') return (dr == 0 || dc == 0) && clearPath(sr, sc, er, ec);
        if (type == 'B') return Math.abs(dr) == Math.abs(dc) && clearPath(sr, sc, er, ec);
        if (type == 'Q') return (dr == 0 || dc == 0 || Math.abs(dr) == Math.abs(dc)) && clearPath(sr, sc, er, ec);
        if (type == 'K') return Math.abs(dr) <= 1 && Math.abs(dc) <= 1;
        if (type == 'N') return Math.abs(dr) == 2 && Math.abs(dc) == 1 || Math.abs(dr) == 1 && Math.abs(dc) == 2;

        return false;
    }

    private boolean clearPath(int sr, int sc, int er, int ec) {
        int stepR = Integer.compare(er, sr);
        int stepC = Integer.compare(ec, sc);

        int r = sr + stepR;
        int c = sc + stepC;

        while (r != er || c != ec) {
            if (!board[r][c].equals("")) return false;
            r += stepR;
            c += stepC;
        }

        return true;
    }

    public void mouseReleased(MouseEvent e) {}
    public void mouseClicked(MouseEvent e) {}
    public void mouseEntered(MouseEvent e) {}
    public void mouseExited(MouseEvent e) {}
}

File: Chess/v1/Chess.py
import tkinter as tk

TILE = 80

class Chess:
    def __init__(self):
        self.root = tk.Tk()
        self.root.title("Simple Python Chess")
        self.canvas = tk.Canvas(self.root, width=640, height=700)
        self.canvas.pack()

        self.board = [
            ["bR","bN","bB","bQ","bK","bB","bN","bR"],
            ["bP","bP","bP","bP","bP","bP","bP","bP"],
            ["","","","","","","",""],
            ["","","","","","","",""],
            ["","","","","","","",""],
            ["","","","","","","",""],
            ["wP","wP","wP","wP","wP","wP","wP","wP"],
            ["wR","wN","wB","wQ","wK","wB","wN","wR"]
        ]

        self.selected = None
        self.white_turn = True
        self.message = "White to move"

        self.canvas.bind("<Button-1>", self.click)
        self.draw()
        self.root.mainloop()

    def click(self, event):
        col, row = event.x // TILE, event.y // TILE
        if row < 0 or row >= 8 or col < 0 or col >= 8:
            return

        if self.selected is None:
            if self.board[row][col] and self.correct_turn(self.board[row][col]):
                self.selected = (row, col)
        else:
            sr, sc = self.selected
            if self.valid_move(sr, sc, row, col):
                self.board[row][col] = self.board[sr][sc]
                self.board[sr][sc] = ""
                self.white_turn = not self.white_turn
                self.message = "White to move" if self.white_turn else "Black to move"
            else:
                self.message = "Illegal move"

            self.selected = None

        self.draw()

    def correct_turn(self, piece):
        return (self.white_turn and piece[0] == "w") or ((not self.white_turn) and piece[0] == "b")

    def valid_move(self, sr, sc, er, ec):
        piece = self.board[sr][sc]
        target = self.board[er][ec]

        if not piece:
            return False
        if target and target[0] == piece[0]:
            return False

        dr, dc = er - sr, ec - sc
        color, kind = piece[0], piece[1]

        if kind == "P":
            direction = -1 if color == "w" else 1
            start = 6 if color == "w" else 1

            if dc == 0 and dr == direction and not target:
                return True
            if dc == 0 and sr == start and dr == 2 * direction and not target and not self.board[sr + direction][sc]:
                return True
            if abs(dc) == 1 and dr == direction and target:
                return True

        if kind == "R":
            return (dr == 0 or dc == 0) and self.clear_path(sr, sc, er, ec)
        if kind == "B":
            return abs(dr) == abs(dc) and self.clear_path(sr, sc, er, ec)
        if kind == "Q":
            return (dr == 0 or dc == 0 or abs(dr) == abs(dc)) and self.clear_path(sr, sc, er, ec)
        if kind == "K":
            return abs(dr) <= 1 and abs(dc) <= 1
        if kind == "N":
            return (abs(dr), abs(dc)) in [(2, 1), (1, 2)]

        return False

    def clear_path(self, sr, sc, er, ec):
        step_r = (er > sr) - (er < sr)
        step_c = (ec > sc) - (ec < sc)

        r, c = sr + step_r, sc + step_c

        while (r, c) != (er, ec):
            if self.board[r][c]:
                return False
            r += step_r
            c += step_c

        return True

    def draw(self):
        self.canvas.delete("all")

        for r in range(8):
            for c in range(8):
                color = "#f0d9b5" if (r + c) % 2 == 0 else "#b58863"
                self.canvas.create_rectangle(c*TILE, r*TILE, c*TILE+TILE, r*TILE+TILE, fill=color)

                if self.selected == (r, c):
                    self.canvas.create_rectangle(c*TILE+4, r*TILE+4, c*TILE+TILE-4, r*TILE+TILE-4, outline="yellow", width=4)

                piece = self.board[r][c]
                if piece:
                    fill = "white" if piece[0] == "w" else "black"
                    self.canvas.create_text(c*TILE+40, r*TILE+43, text=piece[1], fill=fill, font=("Arial", 34, "bold"))

        self.canvas.create_rectangle(0, 640, 640, 700, fill="black")
        self.canvas.create_text(120, 670, text=self.message, fill="white", font=("Arial", 20, "bold"))

Chess()
